onnxruntime/core/session/inference_session.cc

```diff
@@ -511,6 +511,23 @@ common::Status InferenceSession::RegisterExecutionProvider(const std::shared_ptr
     }
   }
 
+  if (provider_type == onnxruntime::kTensorrtExecutionProvider) {
```

Comment:
We need to avoid EP specific code in the core framework. DML is already a bad example. Let's see if we can think of a good design to achieve this.

Reply:
Since we introduced a new virtual function` GetCustomOpDomainList` in IExecutionProvider, I modify the code in framework to handle custom op registration for all Eps.